h2o-core/src/main/java/water/util/FrameUtils.java

```diff
@@ -950,6 +954,59 @@ public static void delete(Lockable ...frs) {
     }
   }
 
+  /**
+   *
+   * @param frame
+   * @param name name of the fold column
+   * @param nfolds number of folds
+   * @param seed
+   */
+  static public Frame addKFoldColumn(Frame frame, String name, int nfolds, long seed) {
+    Vec foldVec = frame.anyVec().makeZero();
+    frame.add(name, AstKFold.kfoldColumn(foldVec, nfolds, seed == -1 ? new Random().nextLong() : seed));
+    return frame;
+  }
+
+  /**
+   *
+   * @param fr
+   * @param name name of the column we want to know index of
+   * @return
+   */
+  static public int getColumnIndexByName(Frame fr, String name) {
+    String[] names = fr.names();
+
+    int foundIndex = Arrays.asList(names).indexOf(name);
+    if (foundIndex == -1) {
+      throw new IllegalArgumentException(String.format("Column with name `%s` was not found in the Frame with key: %s", name,  fr._key));
+    }
+    return foundIndex;
+  }
+
+  /**
+   *
+   * @param frame
+   * @param columnName column's name to be factorized
+   * @return Frame with factorized column
+   */
+  static public Frame asFactor(Frame frame, String columnName) {
```

Comment:
there is one on Frame#toCategoricalVec

Reply:
@michalkurka There is a method like this but only in Vec class. So the only way how I can replace my asFactor method is:
```
fr.replace(fr.find("ColA"), fr.vec("ColA").toCategoricalVec());
```
Maybe we can keep that asFactor method in FrameUtils with above mentioned implementation? 
Is this FrameUtils class from the same category of classes as Frame.java?

Reply:
I think a method that modifies the Vec in place without a copy is safe to use only in certain scenarios. If it was eg. used in an algo (ModelBuilder) it would corrupt the training frame.

Your TE is making a defensive copy of the frame and function like this one is therefore safe in your context. FrameUtils should be safe in any context. I would suggest moving this function closer to the context where it is safe to use, that is your TE implementation.

WDYT?

Reply:
@michalkurka As you mention before, if user don't know what he is doing then he is in trouble. I don't know what is going on in ModelBuilder and if it is safe...but anyway I can move method  to the TE's territory, no problem. But how would you do this transformation yourself? The methods from R/Python clients I think are doing the same replacement as I suggested here:
```
Python:
trainingFrame[targetColumnName] = trainingFrame[targetColumnName].asfactor()
```
Or what is difference?